avm/res/machine-learning-services/workspace/compute/main.bicep

```diff
@@ -0,0 +1,144 @@
+@allowed([
+  'Basic'
+  'Free'
+  'Premium'
+  'Standard'
+])
+param sku string?
+
+@sys.description('Optional. Contains resource tags defined as key-value pairs. Ignored when attaching a compute resource, i.e. when you provide a resource ID.')
+param tags object?
+
+@sys.description('Optional. Flag to specify whether to deploy the compute. Required only for attach (i.e. providing a resource ID), as in this case the operation is not idempotent, i.e. a second deployment will fail. Therefore, this flag needs to be set to "false" as long as the compute resource exists.')
+param deployCompute bool = true
+
+@sys.description('Optional. Location for the underlying compute. Ignored when attaching a compute resource, i.e. when you provide a resource ID.')
+param computeLocation string = resourceGroup().location
+
+@sys.description('Optional. The description of the Machine Learning compute.')
+param description string?
+
+@sys.description('Optional. Opt-out of local authentication and ensure customers can use only MSI and AAD exclusively for authentication.')
+param disableLocalAuth bool = false
+
+@sys.description('Optional. ARM resource ID of the underlying compute.')
+param resourceId string?
+
+@sys.description('Required. Set the object type.')
+@allowed([
+  'AKS'
+  'AmlCompute'
+  'ComputeInstance'
+  'Databricks'
+  'DataFactory'
+  'DataLakeAnalytics'
+  'HDInsight'
+  'Kubernetes'
+  'SynapseSpark'
+  'VirtualMachine'
+])
+param computeType string
+
+@sys.description('Optional. The properties of the compute. Will be ignored in case "resourceId" is set.')
+param properties object?
+
+@sys.description('Optional. The managed identity definition for this resource.')
+param managedIdentities managedIdentitiesType
+
+// ================//
+// Variables       //
+// ================//
+
+var formattedUserAssignedIdentities = reduce(map((managedIdentities.?userAssignedResourceIds ?? []), (id) => { '${id}': {} }), {}, (cur, next) => union(cur, next)) // Converts the flat array to an object like { '${id1}': {}, '${id2}': {} }
+
+var identity = !empty(managedIdentities) ? {
+  type: (managedIdentities.?systemAssigned ?? false) ? (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'SystemAssigned,UserAssigned' : 'SystemAssigned') : (!empty(managedIdentities.?userAssignedResourceIds ?? {}) ? 'UserAssigned' : null)
+  userAssignedIdentities: !empty(formattedUserAssignedIdentities) ? formattedUserAssignedIdentities : null
+} : null
+
+// ============================= //
+// Existing resources references //
+// ============================= //
+
+resource machineLearningWorkspace 'Microsoft.MachineLearningServices/workspaces@2022-10-01' existing = {
+  name: machineLearningWorkspaceName
+}
+
+// ============ //
+// Deployments  //
+// ============ //
+
+resource machineLearningWorkspaceCompute 'Microsoft.MachineLearningServices/workspaces/computes@2022-10-01' = if (deployCompute == true) {
```

Comment:
I'd probably suggest to rename `machineLearningWorkspaceCompute` to just `compute` in alignment with how it's handled for other resources

Reply:
done: https://github.com/Azure/bicep-registry-modules/pull/1055/commits/31ca316848a9674042a411cb588468893f253f3e